### ISSUE TITLE
Helm template cmd  does not support .Capabilities.APIVersions.Has

### ISSUE THREAD

**Issue by kramvan1** [kubernetes-incubator/service-catalog#2604]
# Bug Report 

**What happened**:
https://github.com/kubernetes-incubator/service-catalog/blob/master/charts/catalog/templates/apiregistration.yaml#L7-L10

helm template cmd does not support this. https://github.com/helm/helm/issues/3377
 
**What you expected to happen**:

Allow default either in values.yaml or just pick one and doc the kube version you need.  Just add additional  `else if` to allow the default to take effect.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- service-catalog version:
- Cloud provider or hardware configuration:
- Do you have api aggregation enabled? 
  - Do you see the configmap in kube-system? 
  - Does it have all the necessary fields?
    - `kubectl get cm -n kube-system extension-apiserver-authentication -o yaml` and look for `requestheader-XXX` fields
- Install tools:
  - Did you use helm? What were the helm arguments? Did you `--set` any extra values?
- Are you trying to use ALPHA features? Did you enable them?

**Comment by jberkhahn** [comment]
This is a bug with `template` - these values are intended to be filled in with information from the kube server when you use these templates to `helm install`.

**Comment by kramvan1** [comment]
If you have a supported Minimum kube version (1.12) for this, it would make fixing this very easy:
```
{{- if .Capabilities.APIVersions.Has "apiregistration.k8s.io/v1beta1" }}
apiVersion: apiregistration.k8s.io/v1beta1
{{- else if .Capabilities.APIVersions.Has "apiregistration.k8s.io/v1alpha1" }}
apiVersion: apiregistration.k8s.io/v1alpha1
{{- else }}
apiVersion: apiregistration.k8s.io\/v1beta1
{{- end }}
```

**Comment by jberkhahn** [comment]
this was fixed in Dug's #2609, right? Closing.

**Comment by hickeyma** [comment]
FYI, helm template cmd now supports passing `Capabilities.APIVersions`: helm/helm#3377
